### PR TITLE
Removed ViewComponents and made the list as a partial using a control…

### PR DESCRIPTION
**Pull request by OlSarLang**
…ler instead to manage all users.

### CODE REVIEW

File: Core/Services/IUserService.cs [hunk 1]
```diff
@@ -9,8 +9,9 @@ namespace LexiconLMS.Core.Services
 {
     public interface IUserService
     {
-        Task<SystemUser> GetUserAsync(string? id);
+        Task<SystemUser> GetUserAsync(string id);
```
Comment (Rocangus) [inline Core/Services/IUserService.cs:12]: Good, this got rid of a compile time warning

File: Core/Services/UserService.cs [hunk 2]
```diff
@@ -37,5 +38,16 @@ public async Task<IEnumerable<SystemUserViewModel>> GetUsersViewModelAsync()
...
+
+        public async Task<IEnumerable<SystemUserViewModel>> Filter(string userName)
+        {
+            return await _context.SystemUsers.Where(user => user.Name.ToLower().Contains(userName.ToLower())).Select(user => new SystemUserViewModel
+            {
+                Name = user.Name,
+                Email = user.Email,
+                Id = user.Id,
+                PhoneNumber = user.PhoneNumber
+            }).ToListAsync();
```
Comment (Rocangus) [inline Core/Services/UserService.cs:50]: ToListAsync? Not FirstOrDefaultAsync? There should just be one hit.

File: Data/ApplicationDbContext.cs [hunk 3]
```diff
@@ -22,6 +22,20 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
...
+        {
+            base.OnModelCreating(builder);
+
+            builder.Entity<SystemUserCourse>()
+                .HasKey(k => new
+                {
+                    k.SystemUserId,
+                    k.CourseId
+                });
+        }
```
Comment (Rocangus) [inline Data/ApplicationDbContext.cs]: This will cause issues if we make another migration with this in.

File: Data/ApplicationDbContext.cs [hunk 4]
```diff
@@ -22,6 +22,20 @@ public ApplicationDbContext(DbContextOptions<ApplicationDbContext> options)
...
+
+        public DbSet<LexiconLMS.Core.ViewModels.SystemUserViewModel> SystemUserViewModel { get; set; }
```
Comment (Rocangus) [inline Data/ApplicationDbContext.cs]: This shouldn't be in the database, should it?

**Review by Rocangus — CHANGES_REQUESTED** [review]
There are issues that need solving in ApplicationDbContext that need solving before this can be merged.

File: Core/Services/UserService.cs (hunk 2 above)
Comment (OlSarLang) [reply Core/Services/UserService.cs:50]: It searches for all users containing the search, not just for one.

File: Data/ApplicationDbContext.cs (hunk 4 above)
Comment (OlSarLang) [reply Data/ApplicationDbContext.cs]: woopsie

**Review by Rocangus — APPROVED** [review]
Okay, this seems good now!

**Pull request merged**